errors/middleware-upgrade-guide.md

```diff
@@ -0,0 +1,290 @@
+# Middleware Upgrade Guide
+
+As we work on improving Middleware for General Availability (GA), we've made some changes to the Middleware APIs (and how you define Middleware in your application) based on your feedback.
+
+This upgrade guide will help you understand the changes and how to migrate your existing Middleware to the new API. This guide is for Next.js customers who:
+
+- Currently use the beta Next.js Middleware features
+- Choose to upgrade to the next stable version of Next.js
+
+> **Note:** For customers using Next.js on Vercel, your existing deploys using Middlware will continue to work. Existing sites can be redeployed without needing to upgrade. Before upgrading to the next stable version of Next.js, you will need to follow the upgrade guide below.
```

Comment:
Is there value in being explicit about adding the version of Next.js here so people can be super sure if they have to take action and when? 

Also, typo and text suggestion

```suggestion
> **Note:** For customers using Next.js on Vercel, your existing deploys using Middleware will continue to work. Existing sites can be redeployed without needing to upgrade. Before upgrading to the next stable version of Next.js, you will need to follow this upgrade guide in order for your existing deploy to continue to work.
```

Reply:
Replacing with https://github.com/vercel/next.js/pull/37382/files#r888000812